lib/i18n/backend/http.rb

```diff
@@ -123,6 +131,37 @@ def path(locale)
       def lookup_key(translations, key)
         translations[key]
       end
+
+      def with_retry
+        open_tries ||= 0
+        read_tries ||= 0
+        yield
+      rescue Faraday::ConnectionFailed => e
+        raise unless e.instance_variable_get(:@wrapped_exception).is_a?(Net::OpenTimeout)
+        raise if (open_tries += 1) > @options[:http_open_retries]
+        record :open_retry
+        retry
+      rescue Faraday::TimeoutError => e
+        raise unless e.instance_variable_get(:@wrapped_exception).is_a?(Net::ReadTimeout)
+        raise if (read_tries += 1) > @options[:http_read_retries]
+        record :read_retry
+        retry
+      end
+
+      def record(event, options = {})
+        return unless statsd = @options[:statsd_client]
+
+        case event
+        when :download_fail
+          statsd.increment("#{STATS_NAMESPACE}.download_failed", tags: options[:tags])
+        when :open_retry
+          statsd.increment("#{STATS_NAMESPACE}.open_retry", tags: options[:tags])
+        when :read_retry
+          statsd.increment("#{STATS_NAMESPACE}.read_retry", tags: options[:tags])
```

Comment:
kinda nice to have the raise in the else case, so we can't report random stats. Could check against a set of known things I guess